services/self-service/src/main/java/com/epam/dlab/backendapi/service/impl/OdahuServiceImpl.java

```diff
@@ -132,14 +134,20 @@ public void stop(String name, String project, String endpoint, UserInfo user) {
 
     @Override
     public void terminate(String name, String project, String endpoint, UserInfo user) {
-        Optional<OdahuDTO> odahuDTO = get(project, endpoint);
-        if (odahuDTO.isPresent() && UserInstanceStatus.RUNNING == odahuDTO.get().getStatus()) {
-            odahuDAO.updateStatus(name, project, endpoint, UserInstanceStatus.TERMINATING);
-            actionOnCloud(user, TERMINATE_ODAHU_API, name, project, endpoint);
-        } else {
-            log.error("Cannot terminate odahu cluster {}", odahuDTO);
-            throw new DlabException(String.format("Cannot terminate odahu cluster %s", odahuDTO));
-        }
+        List<OdahuDTO> odahuDTOS = findOdahu();
```

Comment:
Can we use the method which returns not all Odahu clusters, but only per **name** and per ** project**?